account_invoice_analytic_search/models/account_invoice.py

```diff
@@ -0,0 +1,67 @@
+# -*- coding: utf-8 -*-
+# Copyright 2014-2017 Eficent Business and IT Consulting Services S.L.
+#   (http://www.eficent.com)
+# Copyright 2017 Serpent Consulting Services Pvt. Ltd.
+#   (<http://www.serpentcs.com>)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+
+from openerp import fields, models, api
+
+
+class AccountInvoice(models.Model):
+
+    _inherit = "account.invoice"
+
+    @api.depends('invoice_line_ids')
+    def _compute_analytic_accounts(self):
+        res = {}
+        for invoice in self:
+            for invoice_line in invoice.invoice_line_ids:
+                if invoice_line and invoice_line.account_analytic_id:
+                    res[invoice.id].append(invoice_line.account_analytic_id.id)
+        return res
+
+    @api.depends('invoice_line_ids')
+    def _compute_analytic_account_user_ids(self):
+        res = {}
+        for invoice in self:
+            res[invoice.id] = []
+            for invoice_line in invoice.invoice_line_ids:
+                if invoice_line.account_analytic_id and\
+                        invoice_line.account_analytic_id.user_id:
+                    res[invoice.id].append(
+                        invoice_line.account_analytic_id.user_id
+                        )
+        return res
```

Comment:
I wonder if it's possible to do something like this:        
for invoice in self:
     invoice.account_analytic_user_ids = invoice.mapped('invoice.invoice_line_ids.account_analytic_id.user_id')